From f99db682dc7f2e82d03e8596b4dbea98fd73fa9e Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:11:42 -0400
Subject: [powerpc] Fix error return in rtas_flash module init

Message-id: <1410545655-205645-474-git-send-email-dzickus@redhat.com>
Patchwork-id: 94564
O-Subject: [RHEL7 PATCH 473/626] powerpc: Fix error return in rtas_flash module init
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 0c930692107be4b91c3eeecfd9c9d600039f966f
Author: Anton Blanchard <anton@samba.org>
Date:   Mon Apr 14 21:23:32 2014 +1000

    powerpc: Fix error return in rtas_flash module init

    module_init should return 0 or a negative errno.

    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/rtas_flash.c b/arch/powerpc/kernel/rtas_flash.c
index 2f3cdb0..658e89d 100644
--- a/arch/powerpc/kernel/rtas_flash.c
+++ b/arch/powerpc/kernel/rtas_flash.c
@@ -705,7 +705,7 @@ static int __init rtas_flash_init(void)
  if (rtas_token("ibm,update-flash-64-and-reboot") ==
          RTAS_UNKNOWN_SERVICE) {
   pr_info("rtas_flash: no firmware flash support\n");
-  return 1;
+  return -EINVAL;
  }
 
  rtas_validate_flash_data.buf = kzalloc(VALIDATE_BUF_SIZE, GFP_KERNEL);
-- 
1.7.1